core/dbt/parser/schemas.py

```diff
@@ -905,11 +924,12 @@ def parse_patch(self, block: TargetBlock[NodeTarget], refs: ParserRef) -> None:
             if node.patch_path:
                 package_name, existing_file_path = node.patch_path.split("://")
                 raise_duplicate_patch_name(patch, existing_file_path)
-            source_file.append_patch(patch.yaml_key, unique_id)
-            # If this patch has config changes, re-calculate the node config
-            # with the patch config
-            if patch.config:
-                self.patch_node_config(node, patch)
+
+            source_file.append_patch(patch.yaml_key, node.unique_id)
+            # re-calculate the node config with the patch config.  Always do this
+            # for the case when no config is set to ensure the default of true gets captured
+            self.patch_node_config(node, patch)
```

Comment:
I think we can do "if patch.config:" here too, unless I'm missing something?

Reply:
@gshank  since the enabled config is not required, when it's not set it's the same as setting`enabled: true`.  So partial parsing was missing resetting it to true when the `enabled: false` config was removed.  This is why i ended up adding the tests to `068_partial_parsing`.

It didn't look like the expectation was for the patch to set values for things not required so we need to account for a config not being set here.

Reply:
If the patch setting enabled to false was removed, the node should have been reparsed and it should then be true. I'll take a look at the tests.

Reply:
We're not handling disabled nodes in 'delete_schema_mssa_links'.  Here's some code that makes that work:
```
        # remove elem node and remove unique_id from node_patches
        if elem_unique_id:
            # might have been already removed
            if elem_unique_id in self.saved_manifest.nodes or elem_unique_id in self.saved_manifest.disabled:
                if elem_unique_id in self.saved_manifest.nodes:
                    nodes = [self.saved_manifest.nodes.pop(elem_unique_id)]
                else:
                    # The value of disabled items is a list of nodes
                    nodes = self.saved_manifest.disabled.pop(elem_unique_id)
                # need to add the node source_file to pp_files
                for node in nodes:
                    file_id = node.file_id
                    # need to copy new file to saved files in order to get content
                    if file_id in self.new_files:
                        self.saved_files[file_id] = deepcopy(self.new_files[file_id])
                    if self.saved_files[file_id]:
                        source_file = self.saved_files[file_id]
                        self.add_to_pp_files(source_file)
            # remove from patches
            schema_file.node_patches.remove(elem_unique_id)
```